### PR TITLE
Item controller

### PR DESCRIPTION
**Pull request by victortarn**

### CODE REVIEW

File: app/resources/item.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          200:
+            description: Item with given id
+            schema:
+              type: array
+              items:
+                $ref: '#/definitions/Item'
+          500:
+            description: Error fetching items
+        definitions:
+          Item:
```
Comment (cainsusk) [inline app/resources/item.py:44]: Is the `Item` definition complete ? It doesn't seem to fully describe the `item_fields` definition

File: app/resources/item.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        responses:
+          200:
+            description: Item with given id
+            schema:
+              type: array
+              items:
+                $ref: '#/definitions/Item'
+          500:
+            description: Error fetching items
+        definitions:
```
Comment (cainsusk) [inline app/resources/item.py:78]: I believe you only need to define the `Item` once, and then reference it as you do above (`$ref: ...`)

File: app/resources/items.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
+import json
+import random
+from flask_restful import Resource, abort, marshal_with, reqparse, request, fields
+
+from common.db import execute_sql_query, select_table, check_exists, execute_query
+from resources.item import item_fields
+# item_fields = {
```
Comment (cainsusk) [inline app/resources/items.py:7]: perhaps remove this if its not being used

File: app/resources/items.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        responses:
+          200:
+            description: A list of items
+            schema:
+              type: array
+              items:
+                $ref: '#/definitions/Item'
+          500:
+            description: Error fetching items
+        definitions:
```
Comment (cainsusk) [inline app/resources/items.py:38]: same here with the duplicated definition

File: app/resources/item.py (hunk 2 above)
Comment (cainsusk) [reply app/resources/item.py:78]: and if that specific syntax doesn't work, check out how I referenced the `User` definition in `user.py` and `users.py`

File: app/resources/item.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                description: The item ID.
+              ItemName:
+                type: string
+                description: The name of the item.
+              # Add other properties here
+        """
+        return _delete_by_id(id)
+
+def _select_with_id(id): 
+    return execute_query("SELECT * FROM `Items` WHERE `ID` = %s", [(id,)], False)
```
Comment (cainsusk) [inline app/resources/item.py:93]: perhaps use `execute_sql_query` so simplify

**Review by cainsusk — CHANGES_REQUESTED** [review]
Overall looks good !! there's just a few places that should be changed to keep the code readable as well as some duplicated info in the docs. thanks for following the rest-ful style !!

File: app/resources/items.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+def _post_item(item_type, item_name, edition_number, ages, 
+             number_of_parts, location, ordering_company, test_ID): 
+    return execute_query("""INSERT INTO Items (ID, Status, ItemType, ItemName,
+                          EditionNumber, Ages, NumberOfParts, Location, 
+                         OrderingCompany, TestID) 
+                         VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s) """, 
+                         [(_generate_unique_id(), "1", item_type, item_name, 
+                           edition_number, ages, number_of_parts, location, 
+                           ordering_company, test_ID)], True)
+def _generate_unique_id():
```
[the remark this answers is not in the document]
Comment (victortarn) [reply app/resources/items.py:120]: There are no unique identifiers for the test table so we would need to generate a unique id somehow. I think this way is reliable.

**Pull request closed without merging**